Document the new MITRE ATT&CK Coverage page (#3881)
elastic/security-docs: docs/detections/rules-coverage.asciidoc

[[rules-coverage]]
== MITRE ATT&CK® coverage

:frontmatter-description: Review your current coverage of MITRE ATT&CK® tactics and techniques, based on installed rules.
:frontmatter-tags-products: [security]
:frontmatter-tags-content-type: [how-to]
:frontmatter-tags-user-goals: [manage, analyze, visualize]

> The **MITRE ATT&CK® coverage** page shows which https://attack.mitre.org[MITRE ATT&CK®] adversary tactics and techniques are covered by your installed and enabled detection rules. This includes both Elastic prebuilt rules and custom rules.

Reviewer: ```suggestion
The **MITRE ATT&CK® coverage** page shows which https://attack.mitre.org[MITRE ATT&CK®] adversary tactics and techniques are covered by your active detection rules. This includes both Elastic prebuilt rules and custom rules.
```

Author: I think we need to be extra precise here, to explain exactly which rules are included on this page, and what actions (install and enable) the user needs to do to include rules.

> Mirroring the MITRE ATT&CK® framework, columns represent major tactics, and cells within each column represent a tactic's related techniques. Cells are darker when more rules are enabled for each technique, as indicated in the **Legend** at the top.

Reviewer 2: I guess just one nit: a cell is darker when more rules matching the current filters fall under a given cell. This means:

- if only enabled rules are selected in the filter bar, "Cells are darker when more rules are enabled for each technique"
- if both enabled and disabled rules are selected (or none of them), "Cells are darker when more rules are installed for each technique"

[NOTE]
====

> This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. Additionally, rules must be mapped to the MITRE ATT&CK® version used by {elastic-sec}: `v12.1`. If there are Elastic prebuilt rules you haven't installed yet, or custom rules that you either haven't mapped or are mapped to a deprecated tactic or technique, they will not be included in determining your MITRE ATT&CK® coverage.

Reviewer: ```suggestion
This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. Additionally, rules must be mapped to the MITRE ATT&CK® version used by {elastic-sec}: `v12.1`. Elastic prebuilt rules that are not installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique will not be included.
```

Author: Expert edits for conciseness -- thanks @benironside!

====

[role="screenshot"]
image::images/rules-coverage.png[MITRE ATT&CK® coverage page]

[discrete]
=== Filter rules

Use the drop-down filters at the top of the page to control which of your installed detection rules are included in calculating coverage.

* **Installed rule status**: Select to include **Enabled rules**, **Disabled rules**, or both.

* **Installed rule type**: Select to include **Elastic rules** (prebuilt rules), **Custom rules** (user-created rules), or both.

You can also search for a tactic or technique name, technique number, or rule name in the search bar. The search bar acts as a filter for the coverage grid: only rules matching the search term will be included.

NOTE: Searches for tactics and techniques must match exactly, are case sensitive, and do _not_ support wildcards.

[discrete]
=== Expand and collapse cells

Click **Collapse cells** or **Expand cells** to change how much information the cells display. Cells always include the technique's name and the number of sub-techniques covered by enabled rules. Expand the cells to also display counts of disabled and enabled rules for each technique.

> NOTE: The counts inside cells are affected by how you filter the page. For example, if you filter the **Installed rule status** to only include **Enabled rules**, then all disabled rule counts will be 0 because the filter is not including disabled rules.

Reviewer: ```suggestion
NOTE: The counts inside cells are affected by how you filter the page. For example, if you filter the **Installed rule status** to only include **Enabled rules**, then all disabled rule counts will be 0 because disabled rules are filtered out.
```

> The **MITRE ATT&CK® coverage** page shows which https://attack.mitre.org[MITRE ATT&CK®] adversary tactics and techniques are covered by your installed and enabled detection rules.

Reviewer 3: It also includes custom rules, we should add that

Reviewer 3: By default, we show all enabled rules - prebuilt and custom, but users can add disabled rules to the view, or change filters.

Author: ```suggestion
The **MITRE ATT&CK® coverage** page shows which https://attack.mitre.org[MITRE ATT&CK®] adversary tactics and techniques are covered by your installed and enabled detection rules. This includes both Elastic prebuilt rules and custom rules.
```

Mirroring the MITRE ATT&CK® framework, columns represent major tactics, and cells within each column represent a tactic's related techniques. Cells are darker when more rules are enabled for each technique, as indicated in the **Legend** at the top.

> NOTE: This page only includes the detection rules you currently have installed. If there are Elastic prebuilt rules you haven't installed yet, they will not be included in determining your MITRE ATT&CK® coverage.

Reviewer 3: We also do not show rules that do not have MITRE mappings.

Author: ```suggestion
[NOTE]
====
This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. If there are Elastic prebuilt rules you haven't installed yet, or custom rules you haven't mapped to a tactic, they will not be included in determining your MITRE ATT&CK® coverage.

You can map custom rules to tactics in **Advanced settings** when creating or editing a rule.
====
```

[discrete]
=== Filter rules

Use the drop-down filters at the top of the page to control which of your installed detection rules are included in calculating coverage.

* **Installed rule status**: Select to include **Enabled rules**, **Disabled rules**, or both.

* **Installed rule type**: Select to include **Elastic rules** (prebuilt rules), **Custom rules** (user-created rules), or both.

> You can also search for a tactic or technique name, technique number, or rule name in the search bar. Only rules matching the search term will be considered for rules coverage.

Reviewer 3: This search bar acts as filter for the coverage grid.

Author: ```suggestion
You can also search for a tactic or technique name, technique number, or rule name in the search bar. The search bar acts as a filter for the coverage grid: only rules matching the search term will be included.
```
